### ISSUE TITLE
Binary dependencies block the SDK from being use in serverless applications

### ISSUE THREAD

**Issue by kineticsquid** [watson-developer-cloud/python-sdk#431]
#### Expected behavior
I can `import watson_developer_cloud` as part of a cloud function action

#### Actual behavior
The following error:
```
logs    
0    "2018-03-30T00:58:07.576692667Z stderr: Traceback (most recent call last):"
1    "2018-03-30T00:58:07.576736478Z stderr: File \"pythonrunner.py\", line 87, in run"
2    "2018-03-30T00:58:07.576745349Z stderr: exec(self.fn, self.global_context)"
3    "2018-03-30T00:58:07.57675135Z  stderr: File \"__main__.py\", line 15, in <module>"
4    "2018-03-30T00:58:07.576775125Z stderr: import watson_developer_cloud"
5    "2018-03-30T00:58:07.576782617Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/watson_developer_cloud/__init__.py\", line 29, in <module>"
6    "2018-03-30T00:58:07.576792821Z stderr: from .speech_to_text_v1 import SpeechToTextV1"
7    "2018-03-30T00:58:07.576802889Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/watson_developer_cloud/speech_to_text_v1.py\", line 30, in <module>"
8    "2018-03-30T00:58:07.576809819Z stderr: from watson_developer_cloud.websocket import RecognizeCallback, RecognizeListener"
9    "2018-03-30T00:58:07.576815859Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/watson_developer_cloud/websocket/__init__.py\", line 18, in <module>"
10    "2018-03-30T00:58:07.576822118Z stderr: from .speech_to_text_websocket_listener import RecognizeListener"
11    "2018-03-30T00:58:07.576829411Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/watson_developer_cloud/websocket/speech_to_text_websocket_listener.py\", line 21, in <module>"
12    "2018-03-30T00:58:07.576836827Z stderr: from twisted.internet import ssl, reactor"
13    "2018-03-30T00:58:07.576845027Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/twisted/internet/ssl.py\", line 59, in <module>"
14    "2018-03-30T00:58:07.576851282Z stderr: from OpenSSL import SSL"
15    "2018-03-30T00:58:07.577161245Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/OpenSSL/__init__.py\", line 8, in <module>"
16    "2018-03-30T00:58:07.577176459Z stderr: from OpenSSL import crypto, SSL"
17    "2018-03-30T00:58:07.577182836Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/OpenSSL/crypto.py\", line 12, in <module>"
18    "2018-03-30T00:58:07.577188977Z stderr: from cryptography import x509"
19    "2018-03-30T00:58:07.577194796Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/cryptography/x509/__init__.py\", line 8, in <module>"
20    "2018-03-30T00:58:07.577200986Z stderr: from cryptography.x509.base import ("
21    "2018-03-30T00:58:07.57720806Z  stderr: File \"/action/virtualenv/lib/python3.6/site-packages/cryptography/x509/base.py\", line 16, in <module>"
22    "2018-03-30T00:58:07.577214443Z stderr: from cryptography.x509.extensions import Extension, ExtensionType"
23    "2018-03-30T00:58:07.577220485Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/cryptography/x509/extensions.py\", line 18, in <module>"
24    "2018-03-30T00:58:07.577227333Z stderr: from cryptography.hazmat.primitives import constant_time, serialization"
25    "2018-03-30T00:58:07.577247845Z stderr: File \"/action/virtualenv/lib/python3.6/site-packages/cryptography/hazmat/primitives/constant_time.py\", line 9, in <module>"
26    "2018-03-30T00:58:07.577256409Z stderr: from cryptography.hazmat.bindings._constant_time import lib"
27    "2018-03-30T00:58:07.577263071Z stderr: ImportError: Error loading shared library /action/virtualenv/lib/python3.6/site-packages/cryptography/hazmat/bindings/_constant_time.abi3.so: Exec format error"
```

requirements.txt:
```
requests
watson_developer_cloud
```

#### Steps to reproduce the problem
Create a cloud function action using the Python SDK that includes a call to `import watson_developer_cloud` or update an existing working cloud function. The the create/update works fine. The action fails on invocation.

#### Code snippet (Note: Do not paste your credentials)
```
virtualenv virtualenv
source virtualenv/bin/activate
pip install --upgrade pip
pip install -r requirements.txt

cp action.py __main__.py
zip -r action.zip virtualenv common.py __main__.py
rm __main__.py
rm -rf virtualenv

# Update utils package and environment function
bx wsk package update utils --shared yes
bx wsk action update utils/action ./ghe_webhook.zip --kind python:3 --web true
bx wsk action update utils/action --param workspace_id ${WORKSPACE_ID} --param watson_url https://gateway-s.watsonplatform.net/conversation/api 
bx wsk service bind conversation utils/action --instance "Watson Assistant-a7" --keyname Credentials-Action
rm ghe_webhook.zip
```

#### python sdk version
This fails with 1.2.1 and 1.2.0. I tried 1.1.0 and did not encounter this issue. I did encounter an apparently unrelated issue, on a call to `/message`, a `500` error `ValueError: Required property 'conversation_id' not present in Context JSON`

I also tried only importing Conversation with `from watson_developer_cloud import ConversationV1`. Same error though.

#### python version
python 3.5

**Comment by germanattanasio** [comment]
@kineticsquid where you able to try what @ehdsouza suggested above?

**Comment by kineticsquid** [comment]
@germanattanasio @ehdsouza No, that doesn't help. The virtualenv doc at https://virtualenv.pypa.io/en/stable/userguide/ states for `--system-site-packages`:

>If you build with virtualenv --system-site-packages ENV, your virtual environment will inherit packages from /usr/lib/python2.7/site-packages (or wherever your global site-packages directory is).

Meaning the packages I have in my Python site packages are not added to the virtualenv, rather inherited. The error message I then get on the `import watson_developer_cloud` is `ModuleNotFoundError: No module named 'watson_developer_cloud'`

I think the core problem is that one library, `_constant_time.abi3.so`, that can't be loaded in a virtualenv. I ran into a similar problem attempting to use another library with some compiled pieces and I had to resort to building a docker container and deploy that.

Because the offending library is part of Speech to Text and all I'm interested in is Assistant, can these two be better separated?
